### PR TITLE
add focus search bar, open file, dataset shortcut

### PR DESCRIPTION
**Pull request by NakulManchanda**
Signed-off-by: Nakul Manchanda <nakul.manchanda@ibm.com>

### CODE REVIEW

**Comment by NakulManchanda** [comment]
This PR contains following
1) added three new shortcuts
     a) Alt+O - open directory
     b) Alt+K - open dataset
     c) Alt+S - bring focus to serach bar

2) Also, listed shortcuts alongside menu-item

**Comment by NakulManchanda** [comment]
added one more shortcut 
d) Alt+1 - to bring back the focus to editor

File: webClient/src/app/core/menu-bar/menu-bar.config.ts [hunk 1]
```diff
@@ -142,7 +142,8 @@ export const MENU = [
               action: {
                 internalName: 'createFile'
               },
-              keyMap: '' // [Ctrl+N] won't work due to browser conflicts. Keybindings will need
+              keyMap: '', // [Ctrl+N] won't work due to browser conflicts. Keybindings will need
```
Comment (DivergentEuropeans) [inline webClient/src/app/core/menu-bar/menu-bar.config.ts]: You can remove the [Ctrl + N] comment, no longer relevant. A new comment up at the top would be good, something like "Cross browser research shows ALT + [ ] are the least conflicting combinations, so we use these"

File: webClient/src/app/core/menu-bar/menu-bar.config.ts [hunk 2]
```diff
@@ -173,8 +176,9 @@ export const MENU = [
                 action: {
                     internalName: 'saveFile'
                 },
-                keyMap: '' // [Ctrl+S] won't work due to browser conflicts. Keybindings will need
-            },              // to be rethinked.
+                keyMap: '', // [Ctrl+S] won't work due to browser conflicts. Keybindings will need
```
Comment (DivergentEuropeans) [inline webClient/src/app/core/menu-bar/menu-bar.config.ts]: Is this still relevant. Does Ctrl + S conflict with things? Can we just be brave and use Alt + S, is it worth it? @NakulManchanda @1000TurquoisePogs

File: webClient/src/app/core/menu-bar/menu-bar.config.ts [hunk 3]
```diff
@@ -157,13 +158,15 @@ export const MENU = [
                 name: 'Open Directory',
```
Comment (DivergentEuropeans) [inline webClient/src/app/core/menu-bar/menu-bar.config.ts]: @NakulManchanda  Additionally, now that we have a few different shortcuts, yet no documentation yet... can you please add the corresponding hotkey to the title of the action? Similar to this,
image

This was customly requested by our UX team as well.

Comment (DivergentEuropeans) [reply webClient/src/app/core/menu-bar/menu-bar.config.ts]: Oh sorry, you said you did this.

File: webClient/src/app/core/menu-bar/menu-bar.config.ts (hunk 2 above)
Comment (DivergentEuropeans) [reply webClient/src/app/core/menu-bar/menu-bar.config.ts]: Eh, Ctrl + S is too common, I don't even like this suggestion.

File: webClient/src/app/core/menu-bar/menu-bar.config.ts (hunk 1 above)
Comment (NakulManchanda) [reply webClient/src/app/core/menu-bar/menu-bar.config.ts]: removed old comments, also moved shortcut attribute to existing attribute keymap

**Comment by NakulManchanda** [comment]
> Hi Nakul, can you revisit the CSS to determine what you want to do here?
> I noticed the bullet for showing the current language now is on its own line due to CSS changes:
> image

fixed this css bug, re-organized menu row

**Comment by NakulManchanda** [comment]
@DivergentEuropeans 
- Small menu bar elements are clickable & selectable
  removing tabindex from html is more difficult that i thought, null does not work it just makes it 0 which is worse than -1 

- DOM tree in browser
   **How to check on this?**

- Add hotkey to "New Directory" 
   Just a suggestion - i believe we should remove "New Directory" as menu option, learning from vscode
Also, now we have a legit alternate, right click and Create New Directory
Let me know, if you agree - we can remove this from menu

- hotkey vs tab order 
 search bar find it quiet handy, if we want to really make it keyboard friendly, i know right now we do lost control of hotkeys and we had to click to gain focus and key listeners to work.

**Comment by NakulManchanda** [comment]
@DivergentEuropeans 
DOM tree in browser
ok thanks for pointing out in chat - investigating why its blinking on "group lines" menu element

**Comment by NakulManchanda** [comment]
> * Small menu bar elements are clickable & selectable
 - Will deal this in DOM tree fixes, need hook to adjust styles after they are added to DOM, in my opinion this will have same fix as DOM item below.
> * DOM tree in browser suggests some kind of poor Angular element resource management/recursion happening
 - this is not introduced with this PR
> * Add hotkey to "New Directory"
- Removed this from menu

- Rebased again to remove conflicts
- added Alt+ delete as delete shortcut

Looks good to merge. Let me know if you think otherwise. thanks!

**Comment by NakulManchanda** [comment]
@DivergentEuropeans Agreed
 if we add delete for current selection -  "Delete" should be enough, but it should be part of file explorer repo i guess, 
 so far i have not added any keyboard shortcut on tree, but i have a local branch, will experiment and add that as separate PR in explorer project

Few others required for tree - Delete, Up, Down, Enter to open folder etc

**Comment by NakulManchanda** [comment]
removed alt+del will replace by just "Delete" on tree in a separate PR

**Comment by NakulManchanda** [comment]
@DivergentEuropeans thanks for the clue to fix  menu bar group end highlighting issue.
Added ng-if on that div, hope its good thanks!
Thanks again for thorough review!

**Pull request merged**